tests/robottelo/decorators/test_host.py

```diff
@@ -12,139 +14,197 @@
 
 
 class GetHostOsVersionTestCase(TestCase):
+    """Tests for _get_host_os_version version"""
+
     def setUp(self):
+        """Mocking ssh"""
         self._patcher = mock.patch(
             'robottelo.decorators.host.ssh.command')
         self._command = self._patcher.start()
 
     def tearDown(self):
+        """Stop mock created on setUp method"""
         self._patcher.stop()
 
     def assert_rhel_version(self, ssh_version, parsed_version):
+        """
+        Encapsulate assertion logic regarding host os parsing
+        :param ssh_version: version returned from ssh
+        :param parsed_version: parsed version
+        """
         self._command.return_value.stdout = [ssh_version]
-        self.assertEqual(parsed_version, host._get_host_os_version())
+        self.assertEqual(
+            parsed_version,
+            host._get_host_os_version.__wrapped__()
+        )
         self._command.assert_called_once_with('cat /etc/redhat-release')
 
-    def test_rhel_6(self):
+    def test_rhel_major_version_parsing(self):
+        """Version semantic example: 1.2.3
+        1 is major
+        2 is minor
+        3 is patch
+        """
         self.assert_rhel_version(
             u'Red Hat Enterprise Linux Server release 6 (Maipo)',
             u'RHEL6'
         )
 
-    def test_rhel_7_2(self):
+    def test_rhel_minor_version_parsing(self):
+        """Version semantic example: 1.2.3
+        1 is major
+        2 is minor
+        3 is patch
+        """
         self.assert_rhel_version(
             u'Red Hat Enterprise Linux Server release 7.2 (Maipo)',
             u'RHEL7.2'
         )
 
-    def test_rhel_7_2_1(self):
+    def test_rhel_patch_version_parsing(self):
+        """Version semantic example: 1.2.3
+        1 is major
+        2 is minor
+        3 is patch
+        """
         self.assert_rhel_version(
             u'Red Hat Enterprise Linux Server release 7.2.1 (Maipo)',
             u'RHEL7.2.1'
         )
 
     def test_cache(self):
+        """Test _get_host_os_version() is called only once, caching the host
```

Comment:
A simpler docstring would be """Test _get_host_os_version() calls are cached."""